Sources/Overload/OvEditor/src/OvEditor/Panels/MenuBar.cpp

```diff
@@ -216,8 +219,90 @@ void OvEditor::Panels::MenuBar::CreateSettingsMenu()
 
 void OvEditor::Panels::MenuBar::CreateLayoutMenu() 
 {
-	auto& layoutMenu = CreateWidget<MenuList>("Layout");
-	layoutMenu.CreateWidget<MenuItem>("Reset").ClickedEvent += EDITOR_BIND(ResetLayout);
+	auto& layoutMenuList = CreateWidget<MenuList>("Layout");
+
+	auto& uiManager = *EDITOR_CONTEXT(uiManager);
+	const std::filesystem::path& layoutsPath = uiManager.GetLayoutsPath();
+
+	auto& saveMenuItem = layoutMenuList.CreateWidget<MenuItem>("Save");
+	saveMenuItem.ClickedEvent += EDITOR_BIND(SaveCurrentLayout);
+
+	auto& saveNewMenuList = layoutMenuList.CreateWidget<MenuList>("Save New");
+	auto& layoutInputText = saveNewMenuList.CreateWidget<InputFields::InputText>("Layout Name");
+	layoutInputText.selectAllOnClick = true;
+	layoutInputText.EnterPressedEvent += [&layoutsPath](std::string p_input)
+	{
+		if (p_input.empty())
+			return;
+
+		auto& uiManager = *EDITOR_CONTEXT(uiManager);
+		EDITOR_EXEC(DelayAction(std::bind(&OvUI::Core::UIManager::SaveLayout, &uiManager, layoutsPath / (p_input + ".ini")), 1));
+	};
+
+	auto& loadMenuList = layoutMenuList.CreateWidget<MenuList>("Load");
+
+	loadMenuList.ClickedEvent += [&]
+	{
+		loadMenuList.RemoveAllWidgets();
+
+		for (const auto& entry : std::filesystem::directory_iterator(layoutsPath))
+		{
+			if (entry.is_regular_file() && entry.path().extension() == ".ini")
+			{
+				auto& layoutMenuItem = loadMenuList.CreateWidget<MenuItem>(entry.path().stem().string());
+				layoutMenuItem.name = entry.path().stem().string();
```

Comment:
We could display which layout is currently selected with a checkmark
```suggestion
				layoutMenuItem.name = entry.path().stem().string();
				layoutMenuItem.checkable = true;
				layoutMenuItem.checked = Settings::EditorSettings::LatestLayout.Get() == entry.path().stem().string();
```

![image](https://github.com/user-attachments/assets/e4ad2397-d9d7-48f1-b7b5-c5e7c98c9df2)

Reply:
Also, since with this change we can show the currently selected layout, maybe we should always have a "Default" option in the "Load" list, and the "Default" option would have no contextual menu (so it cannot be deleted)